src/ast-validation.md

```diff
@@ -1,6 +1,37 @@
 # AST Validation
 
-AST validation is the process of checking various correctness properties about
-the AST after macro expansion.
+* [About](#about)
+* [Validation](#validation)
 
-**TODO**: write this chapter. [#656](https://github.com/rust-lang/rustc-dev-guide/issues/656)
+## About
+
+_AST validation_ is a separate AST pass that visits each
+item in the tree and performs simple checks. This pass
+doesn't perform any complex analysis, type checking or
+name resolution.
+
+Before performing any validation, the compiler first expands
+the macros. Then this pass performs validations to check
+that each AST item is in the correct state. And when this pass
+is done, the compiler runs the crate resolution pass.
+
+## Validations
+
+Validations are defined in `AstValidator` class, which 
```

Comment:
```suggestion
Validations are defined in `AstValidator` type, which 
```

Reply:
in addition, I would add a link to https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast_passes/ast_validation/struct.AstValidator.html, which is also useful for detecting if the link breaks in future (with the help of mdbook-linkcheck tool)